hyperapp.d.ts

```diff
@@ -37,7 +46,7 @@ export type Children = VNode | string | number | null
  * @memberOf [VDOM]
  */
 export function h<Attributes>(
-  nodeName: Component<Attributes, any, any> | string,
+  nodeName: Component<Attributes> | string,
```

Comment:
I'm pretty sure this is always a component never a lazy one.

Reply:
I think this code is incorrect. Because, according to the code transpiled from Component.tsx to js by TypeScript, the h function receives LazyComponent.

```javascript
const view = (state, actions) => (h("main", null,
    h(Counter, { count: state.count, up: () => actions.up(2), down: actions.down }),
    h(LazyCounter, { label: "Lazy" })));
```

Considering this fact, I think that the definition here is correct.

```ts
export function h<Attributes>(
  nodeName: Component<Attributes> | LazyComponent<Attributes, any, any> | string,
  attributes?: Attributes,
  ...children: Array<Children | Children[]>
): VNode<Attributes>
```

By the way, the reason why the definition before modification does not become a build error is unknown. I think that probably because the type check of tsx syntax is too loose... :confused:

Reply:
Interesting as i built and tested and it seemed to run correctly. Imay have used different options e.g. target and module but that should not impact.

Reply:
Yes, the test code (`test/ts/Component.tsx`) that exists in the repository certainly will be built without errors in my environment. But this is built without error because it is not because the definition is correct, but unfortunately TypeScript performs only looser type checking against jsx syntax than normal code.

By referring to the result of transpiling by TypeScript, if you create a ts file like this and save it as `test/ts/Component.ts`, you will see that it is an error to call the `h` function .
https://gist.github.com/tetradice/95cb50eb787eb63d37872ab2028883a6

(Actually, even the original hyperapp.d.ts will generate an error, so I'm trying to create a new pull request if SteveALee's pull request is accepted)

Reply:
That's definitely not right, but it did seem to run correctly for me. Hidden bug! Thanks for the gist.

I'll tidy up the PR now and see.

Reply:
> I think that probably because the type check of tsx syntax is too loose

That's interesting and if true pushes one to use hyperscript / functioncalls for good type safety

Reply:
@tetradice I confirm you suggested fix works in your gist.
BUT I got extra null assignment errors as well on other lines. I left details on your gist.